tests/python_package_test/test_dask.py

```diff
@@ -1108,3 +1108,26 @@ def test_dask_methods_and_sklearn_equivalents_have_similar_signatures(methods):
     for param in dask_spec.args:
         error_msg = f"param '{param}' has different default values in the methods"
         assert dask_params[param].default == sklearn_params[param].default, error_msg
+
+
+def test_training_succeeds_when_data_is_dataframe_and_label_is_column_array(
+    client,
+    listen_port
+):
```

Comment:
thanks for this! I think test could be a bit stronger. Instead of just smoke-testing that training runs without error, could you make this test check that `label` vectors of shape `[n_samples, 1]` and `[n_samples]` result in identical models?

I think that could look something like this

```python
    params = {
        'n_estimators': 1,
        'num_leaves': 3,
        'local_listen_port': listen_port,
        'time_out': 5
    }
    _, _, _, dX, dy, dw = _create_data(
        objective='regression',
        output='dataframe'
    )
    dy = dy.to_dask_array(lengths=True)
    dy_col_array = dy.reshape(-1, 1)
    assert len(dy_col_array.shape) == 2 and dy_col_array.shape[1] == 1

    model1 = lgb.DaskLGBMRegressor(**params)
    model1.fit(dX, dy, sample_weight=dw)
    p1 = model1.predict(dX).compute()
    assert model1.fitted_
    
    dy_vector = dy.reshape(-1)
    assert len(dy_col_array.shape) == 1
    model2 = lgb.DaskLGBMRegressor(**params)
    model2.fit(dX, dy, sample_weight=dw)
    p2 = model2.predict(dX).compute()
    assert model2.fitted_
    
    assert_eq(p1, p2)
```

Notice in the above that I also changed the check from `assert model.booster_` to `assert model.fitted_`. I think that's a more direct check of the thing we're trying to check ("is the model fitted").